packages/liveblocks-core/src/room.ts

```diff
@@ -1730,12 +1723,10 @@ function makeStateMachine<
   function onClose(event: { code: number; wasClean: boolean; reason: string }) {
     context.socket = null;
 
-    clearTimeout(context.timeoutHandles.pongTimeout);
-    clearInterval(context.intervalHandles.heartbeat);
-    if (context.timeoutHandles.flush) {
-      clearTimeout(context.timeoutHandles.flush);
-    }
-    clearTimeout(context.timeoutHandles.reconnect);
+    clearTimeout(context.timers.flush);
+    clearTimeout(context.timers.reconnect);
+    clearInterval(context.timers.heartbeat);
```

Comment:
in the original code it was more clear that heartbeat was an interval, and thus needed to be cleared via `clearInterval`. Maybe instead of moving it to timers, put it in a `context.intervals`.

Reply:
Can do, but I think what this will evolve into is some kind of grouping of the heartbeat + pong timers. These two things have a relationship, and the the reconnect timer has a different trigger. In the eventual state machine this will evolve into, I'd love to make the heartbeat and pong handler part of the enter/exit criteria for the "connected" phase. The reconnect timer is part of a different state of the state machine, where the reconnect backoff strategy gets implemented.

So I think no matter how we organize them here now, I think they will eventually get separated again by function more than by the fact that they are timers.